### ISSUE TITLE
Does S3 sync executes copy in parallel?

### ISSUE THREAD

**Issue by yudho** [aws/aws-cli#4340]
Does S3 Sync command execute necessary operations (e.g. cp) in parallel fashion?

**Comment by justnance** [comment]
@yudho - Thanks for asking.  Yes, the S3 Sync command can execute in parallel.  There is more information in our [docs](https://github.com/aws/aws-cli/blob/develop/awscli/topics/s3-config.rst) and a few related issues: #907, #1122, #905, and #658. 

Let us know if this helps.

**Comment by yudho** [comment]
Awesome. Thank you

**Comment by aryus96** [comment]
Hi,

I cannot execute two aws sync operation in parallel. I'm trying to upload two folders to the same S3 Bucket that is in a different region than the EC2 with the files.
I tried with cp aswell.

Two case scenarios :
- The first sync start and upload correctly. The second one is on hold and waiting for the first one to finish. They are both on different ssh session.
- The first sync works, and the second, if I use the cp command, return the following error :

> s3 upload failed need to rewind .... the stream but stream is not seekable
